cli/azd/pkg/input/console.go

```diff
@@ -65,6 +65,8 @@ type Console interface {
 	GetWriter() io.Writer
 	// Gets the standard input, output and error stream
 	Handles() ConsoleHandles
+	// Set formatter to none.
+	RemoveFormatter()
```

Comment:
Can you help me understand how this is used?

Reply:
Based on my conversation with @vhvb1989 , the error output was being `json` formatted which is a poor user experience when working on an application built using `DefaultAzureCredential`.

I had recommended logging an issue so that we can prioritize for fixing, and moving it out of the scope of this change, since the right change is to rethink `console`'s implementation of JSON formatting, and we need to get crisp on that. I think I'd still prefer that.

Reply:
@ellismg , for any command using `--o json`, if the action returns an error within the `ActionResult`, it is converted to `json` like:

![image](https://user-images.githubusercontent.com/24213737/225983524-ff2c0c1e-ad65-41e7-8c04-5a4e0f35a30c.png)

I added this fix to remove any formatter from a console in case of an error:

![image](https://user-images.githubusercontent.com/24213737/225983744-408cbaf1-a5e5-4aeb-b39a-32a03b10dc45.png)

Then the error looks like expected:

![image](https://user-images.githubusercontent.com/24213737/225983806-1826392b-f161-4dcf-8c17-36d32bf6b0dc.png)

Reply:
I'm not sure this is the right direction to be moving.  Naively, I would assume that when you use `-o json` we would produce the error in some structured way that the caller (who is likely a program, not a human, because they passed `-o json`) could easily interpret?

I feel like the first screenshot is closer to where we want to be than the second.  I think in practice what we'd want is something like this:

`azdd auth token -o json` fails and writes a non zero exit code.  To `stdout` it emits something like:

`{ "message": "resolving the Azure Directory ... "}`

But I get that getting there is a separate change where we define a common JSON encoding of our errors are start using it when `--output json` is true.

I think I'd remove this suppression stuff you added for now.  I'm okay with the error UX of `azd auth token --output json` not looking great to humans for now.

Reply:
reverted